pybleau_plotting.puml

```diff
@@ -0,0 +1,144 @@
+@startuml
+
+'class DataElement
+'DataElement <|-- DataFramePlotManager
+'DataElement <|-- DataFrameAnalyzer
+class DataFrameAnalyzer {
+    DataFrame source_df
+    DataFrame filtered_df
+    DataFrame displayed_df
+    DataFrame summary_df
+    DataFrame summary_categorical_df
+    [DataFramePlotManager] plot_manager_list
+}
+DataFrameAnalyzer o--> DataFrame
+DataFrameAnalyzer "0..*" o--> DataFramePlotManager
+
+class DataFramePlotManager {
+    DataFrame data_source
+    [PlotDescriptor] contained_plots
+    DataFrameAnalyzer source_analyzer
+    IPlotTemplateInteractor template_interactor
+    ..Job Description..
+    * Manager to create, update, reorder, hide, delete plots.
+    * Provides a data model for a DataFrameAnalyzer.
+}
+DataFramePlotManager "0..*" o--> PlotDescriptor
+DataFramePlotManager o..> DataFrame
+DataFramePlotManager o--> DataFrameAnalyzer
+
+class BasePlotFactory {
+    ArrayPlotData plot_data
+    BaseXYPlotStyle plot_style
+    PlotContextMenuManager context_menu_manager
+    ..Job Description..
+    * Builds a chaco plot with generate_plot()
+    * Manipulates titles and ticks/labels for a plot.
+}
+BasePlotFactory o--> BaseXYPlotStyle
+BasePlotFactory o--> PlotContextMenuManager
+
+class StdXYPlotFactory {
+    OverlayPlotContainer plot
+    {BaseXYPlot} renderers
+    ..Job Description..
+    * Creates a dict of properties for a 2D plot
+    with one or more renderers and a legend.
+    * Initializes plot data from arrays.
+    * Adds tools (legend/context menu/zoom/pan) to a plot.
+    * Builds and adds/appends/removes/updates renderers
+}
+BasePlotContainer <|-- OverlayPlotContainer
+BasePlotFactory <|-- StdXYPlotFactory
+StdXYPlotFactory o--> OverlayPlotContainer
+OverlayPlotContainer : * Chaco Plot container that stretches all its components
```

Comment:
What do you mean by `stretches`?

Reply:
It stretches all of its components to the bounds it is given, so they are all covering the whole space provided. That's what I understand it to be, anyway. I got that info from the `OverlayPlotContainer` docs.